### PR TITLE
Add File Downloading Capabilities 

### PR DESCRIPTION
**Pull request by cryptidv**
### Background
This command will extend the power and functionality of Auto-GPT by allowing it to download files natively.

### Changes
* Added download_file command to allow Auto-GPT to download files natively, if needed.
* Added `--alow-downloads` flag

### Test Plan
Tested with:
```yaml
ai_goals:
- 'Download this file: "https://images.hdqwalls.com/wallpapers/cute-kitten-4k-im.jpg"
  and save it locally.'
ai_name: DownloaderGPT
ai_role: Downloads files. AND DOES NOTHING ELSE.
```
Stress tested with:
```yaml
ai_goals:
- 'Download this file: "https://download.manjaro.org/kde/22.0.5/manjaro-kde-22.0.5-230316-linux61.iso" and save it locally'
ai_name: DownloaderGPT
ai_role: Downloads files. AND DOES NOTHING ELSE.
```
### PR Quality Checklist
- [x] My pull request is atomic and focuses on a single change.
- [x] I have thoroughly tested my changes with multiple different prompts.
- [x] I have considered potential risks and mitigations for my changes.
- [ ] I have documented my changes clearly and comprehensively.
- [x] I have not snuck in any "extra" small tweaks changes

### CODE REVIEW

**Review by nponeccop — DISMISSED** [review]

File: autogpt/spinner.py [hunk 1]
```diff
@@ -29,12 +29,14 @@ def spin(self) -> None:
             time.sleep(self.delay)
             sys.stdout.write(f"\r{' ' * (len(self.message) + 2)}\r")
 
-    def __enter__(self) -> None:
+    def __enter__(self):
```
Comment (richbeales) [inline autogpt/spinner.py:32]: You appear to have removed a type annotation here?

**Comment by richbeales** [comment]
conflicts with  autogpt/args.py

File: autogpt/spinner.py (hunk 1 above)
Comment (sadmuphin) [reply autogpt/spinner.py:32]: It now returns `self` not None.

**Comment by cryptidv** [comment]
@richbeales Apologies should be resolved now 👍

**Review by nponeccop — DISMISSED** [review]

**Comment by richbeales** [comment]
please follow the new method here (https://github.com/Significant-Gravitas/Auto-GPT/pull/1551) which only adds the command to the prompt if it's 'turned on'.  Same logic should apply here.

**Comment by cryptidv** [comment]
Made a mistake with the commit, will reopen

**Pull request closed without merging**